src/snowflake/cli/plugins/snowpark/commands.py

```diff
@@ -308,31 +314,108 @@ def _deploy_single_object(
     }
 
 
+deprecated_pypi_download_option = typer.Option(
+    YesNoAsk.NO.value,
+    "--pypi-download",
+    help="Whether to download non-Anaconda packages from PyPi.",
+    hidden=True,
+    callback=deprecated_flag_callback_enum(
+        "--pypi-download flag is deprecated. Snowpark build command"
+        " always tries to download non-Anaconda packages from PyPi."
+    ),
+)
+
+
+def _write_requirements_file(file_path: SecurePath, requirements: List[Requirement]):
+    log.info("Writing requirements into file %s", file_path.path)
+    file_path.write_text("\n".join(req.line for req in requirements))
+
+
 @app.command("build")
 @with_project_definition("snowpark")
 def build(
-    pypi_download: YesNoAsk = PyPiDownloadOption,
-    check_anaconda_for_pypi_deps: bool = CheckAnacondaForPyPiDependencies,
-    package_native_libraries: YesNoAsk = PackageNativeLibrariesOption,
+    ignore_anaconda: bool = IgnoreAnacondaOption,
+    allow_shared_libraries: bool = AllowSharedLibrariesOption,
+    index_url: Optional[str] = IndexUrlOption,
+    skip_version_check: bool = SkipVersionCheckOption,
+    deprecated_package_native_libraries: YesNoAsk = deprecated_allow_native_libraries_option(
+        "--package-native-libraries"
+    ),
+    deprecated_check_anaconda_for_pypi_deps: bool = DeprecatedCheckAnacondaForPyPiDependencies,
+    _deprecated_pypi_download: YesNoAsk = deprecated_pypi_download_option,
     **options,
 ) -> CommandResult:
     """
     Builds the Snowpark project as a `.zip` archive that can be used by `deploy` command.
     The archive is built using only the `src` directory specified in the project file.
     """
-    paths = SnowparkPackagePaths.for_snowpark_project(
+    if not deprecated_check_anaconda_for_pypi_deps:
+        ignore_anaconda = True
+    # TODO: yes/no/ask logic should be removed in 3.0
+    allow_shared_libraries_yesnoask = {
+        True: YesNoAsk.YES,
+        False: YesNoAsk.NO,
+    }[allow_shared_libraries]
+    if deprecated_package_native_libraries != YesNoAsk.NO:
+        allow_shared_libraries_yesnoask = deprecated_package_native_libraries
```

Comment:
Let's try to remove this piece of code by using `or` in if
```py
if package_utils.detect_and_log_shared_libraries(
                download_result.downloaded_packages_details
            )
     if not (allow_shared_libraries or resolve_allow_shared_libraries_yes_no_ask(...)):
         raise ClickException(
                    "Some packages contain shared (.so/.dll) libraries. "
                    "Try again with --allow-shared-libraries."
                )
```
WDYT?

Reply:
Makes sense - I decided to merge them as they were passed as a single argument to the `download_unavailable_packages` function, but that's not a case anymore